contracts/contracts/harvest/Harvester.sol

```diff
@@ -0,0 +1,376 @@
+     */
+    function _harvest() internal {
+        address[] memory allStrategies = IVault(vaultAddress)
+            .getAllStrategies();
+        for (uint256 i = 0; i < allStrategies.length; i++) {
+            _harvest(allStrategies[i]);
+        }
+    }
+
+    /**
+     * @dev Collect reward tokens for a specific strategy. Called from the vault.
+     * @param _strategyAddr Address of the strategy to collect rewards from
+     */
+    function harvest(address _strategyAddr) external onlyGovernor nonReentrant {
+        // TODO: add protection so that harvestAndSwap isn't called twice too closely together
+        _harvest(_strategyAddr);
+    }
+
+    /**
+     * @dev Collect reward tokens for a specific strategy and swap for supported
+     *      stablecoin via Uniswap. Called from the vault.
+     * @param _strategyAddr Address of the strategy to collect rewards from
+     */
+    function harvestAndSwap(address _strategyAddr) external nonReentrant {
+        IStrategy strategy = IStrategy(_strategyAddr);
+        _harvest(address(strategy));
+        address[] memory rewardTokens = strategy.getRewardTokenAddresses();
+        for (uint256 i = 0; i < rewardTokens.length; i++) {
+            _swap(rewardTokens[i]);
+        }
+        IVault(vaultAddress).rebase();
+    }
+
+    /**
+     * @dev Collect reward tokens from a single strategy and swap them for a
+     *      supported stablecoin via Uniswap
+     * @param _strategyAddr Address of the strategy to collect rewards from.
+     */
+    function _harvest(address _strategyAddr) internal {
+        IStrategy strategy = IStrategy(_strategyAddr);
+        address[] memory rewardTokenAddresses = strategy
+            .getRewardTokenAddresses();
+
+        if (rewardTokenAddresses.length > 0) {
+            strategy.collectRewardTokens();
+        }
+    }
+
+    /**
+     * @dev Swap all supported swap tokens for stablecoins via Uniswap.
+     */
+    function _swap() internal {
+        address[] memory allStrategies = IVault(vaultAddress)
+            .getAllStrategies();
+
+        for (uint256 i = 0; i < swapTokens.length; i++) {
+            _swap(swapTokens[i]);
+        }
+    }
+
+    /**
+     * @dev Swap a record token for stablecoins for Uniswap. The token must have
+     *       a registered price feed with the price provider.
+     * @param _swapToken Address of the token to swap.
+     */
+    function _swap(address _swapToken)
+        internal
+        returns (uint256[] memory swapResult)
+    {
+        RewardTokenConfig storage tokenConfig = rewardTokenConfigs[_swapToken];
+        require(
+            tokenConfig.uniswapV2CompatibleAddr != address(0),
+            "Swap token is missing token configuration."
+        );
+
+        address priceProvider = IVault(vaultAddress).priceProvider();
+        address[] memory allAssets = IVault(vaultAddress).getAllAssets();
+
+        if (tokenConfig.uniswapV2CompatibleAddr != address(0)) {
```

Comment:
Don't we just check this in a require statement a few lines above?

Sometimes we may want to disable selling a token perhaps by setting the address to zero.

I think we should just return if the address is zero, in the place we currently have the require statement.

Reply:
Thanks removed the double check. And added comments to `uniswapV2CompatibleAddr` property that setting to zero address disables swaps.